GitHubCodespaces.ps1

```diff
@@ -0,0 +1,538 @@
+
+            break
+        }
+
+        'CodespaceName' {
+            $telemetryProperties['CodespaceName'] = Get-PiiSafeString -PlainText $CodespaceName
+
+            $uriFragment = "user/codespaces/$CodespaceName"
+            $description = "Getting user/codespaces/$CodespaceName"
+
+            break
+        }
+
+        'Organization' {
+            # /orgs/{org}/codespaces
+            # /orgs/{org}/members/{username}/codespaces
+
+            $telemetryProperties['OrganizationName'] = Get-PiiSafeString -PlainText $OrganizationName
+            if ([string]::IsNullOrWhiteSpace($UserName)) {
+                $uriFragment = "orgs/$OrganizationName/codespaces"
+                $description = "Getting codespaces for $OrganizationName"
+            }
+            else {
+                $telemetryProperties['UserName'] = Get-PiiSafeString -PlainText $UserName
+                $uriFragment = "orgs/$OrganizationName/members/$UserName/codespaces"
+                $description = "Getting codespaces for $OrganizationName"
+            }
+
+            break
+        }
+
+        'Repository' {
+            $elements = Resolve-RepositoryElements
+            $OwnerName = $elements.ownerName
+            $RepositoryName = $elements.repositoryName
+
+            $telemetryProperties['OwnerName'] = Get-PiiSafeString -PlainText $OwnerName
+            $telemetryProperties['RepositoryName'] = Get-PiiSafeString -PlainText $RepositoryName
+
+            $uriFragment = "repos/$OwnerName/$RepositoryName/codespaces"
+            $description = "Getting $OwnerName/$RepositoryName/codespaces"
+
+            break
+        }
+
+        'Uri' {
+            $elements = Resolve-RepositoryElements
+            $OwnerName = $elements.ownerName
+            $RepositoryName = $elements.repositoryName
+
+            $telemetryProperties['OwnerName'] = Get-PiiSafeString -PlainText $OwnerName
+            $telemetryProperties['RepositoryName'] = Get-PiiSafeString -PlainText $RepositoryName
+
+            $uriFragment = "repos/$OwnerName/$RepositoryName/codespaces"
+            $description = "Getting $OwnerName/$RepositoryName/codespaces"
+
+            break
+        }
+    }
+
+    $params = @{
+        'UriFragment' = $uriFragment
+        'Description' = $description
+        'AcceptHeader' = "$script:nebulaAcceptHeader,$script:baptisteAcceptHeader,$script:mercyAcceptHeader"
+        'AccessToken' = $AccessToken
+        'TelemetryEventName' = $MyInvocation.MyCommand.Name
+        'TelemetryProperties' = $telemetryProperties
+    }
+
+    $result = Invoke-GHRestMethodMultipleResult @params
+    if ($result.codespaces) {
+        $result = $result.codespaces
+    }
+
+    return ($result | Add-GitHubCodespaceAdditionalProperties)
+}
+
+function Start-GitHubCodespace
+{
+<#
+    .SYNOPSIS
+        Start a user's codespace.
+
+    .DESCRIPTION
+        You must authenticate using an access token with the codespace scope to use this endpoint.
```

Comment:
These first two lines in the description are great entries for a `.NOTES` section at the end of this documentation section.  For now, please just duplicate your `.SYNOPSIS` here, unless you have additional details on what this is doing (as opposed to requirements for using it).